File: ms_relatorio/src/main/java/com/saude_mais/ms_relatorio/entities/Paciente.java
package com.saude_mais.ms_relatorio.entities;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Paciente {

    private String id;
    private String nomePac;
    private String tel;
    private String dtNasci;
    private String ocupacao;
    private List<Consulta> consulta;

}

File: validationSchema/ms_gerenciamento/paciente.js
use clinica;

db.createCollection("paciente", {
    validator: {
        $jsonSchema: {
            required: ["nomePac", "tel", "dtNasci", "ocupacao", "consulta"],
            properties: {
                nomePac: {
                    bsonType: "string",
                    description: "nome do paciente"
                },
                tel: {
                    bsonType: "string",
                    maxLength: 11,
                    description: "telefone do paciente"
                },
                dtNasci: {
                    bsonType: "string",
                    maxLength: 10,
                    description: "'data de nascimento do paciente"
                },
                ocupacao: {
                    bsonType: "string",
                    maxLength: 30,
                    description: "ocupação do paciente"
                },
            }, consulta: {
                bsonType: "array", description: "consultas do paciente.", minItems: 1,
            }
        }
    }
})


